nmdc_server/api.py

```diff
@@ -747,6 +754,116 @@ async def update_submission(
     return submission
 
 
+def create_github_issue(submission, user):
+    settings = Settings()
+    gh_url = str(settings.github_issue_url)
+    token = settings.github_authentication_token
+    # If the settings for issue creation weren't supplied return, no need to do anything further
+    if gh_url == None or token == None:
+        return
+
+    # Gathering the fields we want to display in the issue
+    headers = {"Authorization": f"Bearer {token}", "Content-Type": "text/plain; charset=utf-8"}
+    studyform = submission.metadata_submission["studyForm"]
+    contextform = submission.metadata_submission["contextForm"]
+    multiomicsform = submission.metadata_submission["multiOmicsForm"]
+    pi = studyform["piName"]
+    piorcid = studyform["piOrcid"]
+    datagenerated = "Yes" if contextform["dataGenerated"] else "No"
+    omicsprocessingtypes = ", ".join(multiomicsform["omicsProcessingTypes"])
+    sampletype = ", ".join(submission.metadata_submission["templates"])
+    sampledata = submission.metadata_submission["sampleData"]
+    numsamples = 0
+    for key in sampledata:
+        numsamples = max(numsamples, len(sampledata[key]))
+
+    # some variable data to supply depending on if data has been generated or not
+    data_ids = []
+    if contextform["dataGenerated"]:
+        data_ids = [
+            "NCBI ID: " + multiomicsform["NCBIBioProjectId"],
+            "GOLD ID: " + multiomicsform["GOLDStudyId"],
+            "Alternative IDs/Names: " + ", ".join(multiomicsform["alternativeNames"]),
+        ]
+
+    else:
+        data_ids = [
+            "JGI IDs: " + multiomicsform["JGIStudyId"],
+            "EMSL IDs: " + multiomicsform["studyNumber"],
+            "Alternative IDs/Names: " + ", ".join(multiomicsform["alternativeNames"]),
+        ]
+
+    # assemble the body of the API request
+    body_lis = [
+        f"Submitter: {user.name}, {user.orcid}",
+        f"Submission ID: {submission.id}",
+        f"Has data been generated: {datagenerated}",
+        f"PI name: {pi}",
+        f"PI orcid: {piorcid}",
+        "Status: Submitted -Pending Review",
+        f"Data types: {omicsprocessingtypes}",
+        f"Sample type:{sampletype}",
+        f"Number of samples:{numsamples}",
+        data_ids[0],
+        data_ids[1],
+        data_ids[2],
+        "Note:",
+    ]
+    body_string = " \n ".join(body_lis)
+    payload_dict = {
+        "title": f"NMDC Submission: {submission.id}",
+        "body": body_string,
+        "assignees": ["mslarae13"],
```

Comment:
Can the assignee come from `Settings` as well? I'm just thinking that if someone wants to test this locally they probably don't want to spam Montana with issue assignments.

Reply:
That's a good point. I'll add that in.